src/libs/ajax/leonardo/models/core-models.ts

```diff
@@ -10,6 +10,10 @@ export interface AuditInfo {
   dateAccessed: string;
 }
 
+export interface IHaveCreator {
```

Comment:
Actually... why do we even need this type? Environments passes the disk or runtime to the permissions functions, not a slice of those objects.

Reply:
it's to narrow the required object shape so that any of the three can work (App, Runtime, Disk)

Reply:
Why create a narrower type vs using a union type to more precisely reflect what's passed?

Reply:
following the I in SOLID, favor smaller interfaces over larger ones.  The narrower interface makes the permission provider more flexible since it can take anything that has thing.auditInfo.creator.... if needs change later, we can broaden the type.

Reply:
hmmm.... I get your logic.... How about I tweak it so that HasCreator is typed { creator: string } and the component can pass in resource.auditInfo?

Reply:
But also, the more I think about it, the more I think this might be the wrong abstraction... The component doesn't need to ask the app to determine if the user has permission to access a resource. It _knows_ how to determine that. It only needs to know the user's email address. As-is, each app has to duplicate the same logic in the permissions provider.

Reply:
yes, but I think there is value in separating the concern completely... this has the benefit of later centralizing and managing change in the newly isolated provided versus hunting down usage in various components.

Reply:
ok, I tried out having it take in just the audit object with only creator field specified, but don't like how it feels... forces consumer to be more verbose... permissions.canDeleteDisk(disk.auditInfo) feels clunky compared to just permissions.canDeleteDisk(disk)...

So I think I will go back to the earlier suggestion of just passing in the whole object/resource, and typing as union accordingly.